File: takeMeFront/takeMeFront/src/types/Order.ts
/**
 * 订单（老人结算后生成）
 * 作用：记录一次完整购买
 */
export interface Order {
  id: number                      // 订单主键
  orderNo: string                 // 订单编号（给用户看）
  userId: number                  // 下单老人ID
  volunteerId?: number            // 接单志愿者ID

  totalPrice: number              // 订单总金额

  serviceDate: string             // 服务日期
  serviceTime: string             // 服务时间
  address: string                 // 服务地址
  remark?: string                 // 备注

  status: 0 | 1 | 2 | 3 | 4 | 5   // 0待接单 1已接单 2服务中 3待确认 4已完成 5已取消
  createTime: string              // 下单时间
  completeTime?: string           // 完成时间
  isReviewed: boolean             // 是否已评价
}

File: takeMe/pojo/src/main/java/com/me/entity/Order.java
package com.me.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("`order`")
public class Order {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String orderNo;
    private Long userId;
    private String volunteerIds;

    private Integer totalPrice;
    private String serviceDate;
    private String serviceTime;
    private String address;
    private String remark;

    /**
     * 0: 待接单 - 刚创建或所有服务项都被放弃
     * 1: 已接单 - 至少有一个服务项被接单
     * 2: 服务中 - 至少有一个服务项正在服务中
     * 3: 待确认 - 所有服务项都已完成(状态≥3)
     * 4: 已完成 - 用户确认完成
     * 5: 已取消 - 用户/系统取消
     * 6: 模拟未支付
     */
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime completeTime;
    private Integer isReviewed;
}